crates/sequencer/src/transaction.rs

```diff
@@ -0,0 +1,61 @@
+use std::sync::Arc;
+
+use blockifier::transaction::{
+    account_transaction::AccountTransaction,
+    transaction_execution::Transaction as ExecutionTransaction,
+};
+use starknet::core::types::{BroadcastedTransaction, FieldElement};
+use starknet_api::core::{ContractAddress, Nonce, PatriciaKey};
+use starknet_api::hash::{StarkFelt, StarkHash};
+use starknet_api::transaction::InvokeTransaction;
+use starknet_api::transaction::{
+    Calldata, Fee, InvokeTransactionV1, TransactionHash, TransactionSignature,
+};
+
+#[derive(Debug)]
+pub struct StarknetTransaction(BroadcastedTransaction);
+
+impl StarknetTransaction {
+    pub fn new(transaction: BroadcastedTransaction) -> Self {
+        Self(transaction)
+    }
+}
+
+impl TryFrom<StarknetTransaction> for ExecutionTransaction {
+    type Error = eyre::Error;
+
+    fn try_from(transaction: StarknetTransaction) -> Result<Self, Self::Error> {
+        match transaction.0 {
+            BroadcastedTransaction::Invoke(invoke) => Ok(ExecutionTransaction::AccountTransaction(
+                AccountTransaction::Invoke(InvokeTransaction::V1(InvokeTransactionV1 {
+                    transaction_hash: TransactionHash(Into::<StarkHash>::into(
+                        Into::<StarkFelt>::into(FieldElement::ONE), // TODO: Replace with actual computed hash.
```

Comment:
is there a risk to have a weird bug somewhere because of this? it is difficult to put the real hash?

Reply:
I don't think we ever use the transaction hash, and it is [not complicated](https://docs.starknet.io/documentation/architecture_and_concepts/Network_Architecture/transactions/#v1_hash_calculation) but just adds some overhead which I don't think is very useful right now

Reply:
I'm just scared that we forgot this and one day face a bugs. I prefer if we put the real hash